src/PrestaShopBundle/Controller/Admin/Improve/Payment/PaymentPreferencesController.php

```diff
@@ -77,8 +104,8 @@ public function indexAction(Request $request)
     }
 
     /**
-     * Process payment modules preferences form.
-     *
+     * @deprecated should be removed in next major version in favor of multiple separate actions
```

Comment:
It might be some special case, but the deprecation is incorrect :thinking: 
Usually we implement new method, use it in core, and deprecate the old one (else its not a deprecation, but just some promise to do something. Even the keyword "should" suggests that you are not sure it will be done in next major, so Im abit confused about this stuff.). Can't we work around this method to make a correct deprecation?

Reply:
or are those actions the replacement `processPaymentCarrierRestrictionsFormAction`, `processPaymentCountryRestrictionsFormAction`, `processPaymentCurrencyRestrictionsFormAction`, `processPaymentGroupRestrictionsFormAction` ? The deprecation should look like:
```suggestion
     * @deprecated will be removed in next major version. Use following separate actions instead:
     * @see processPaymentCarrierRestrictionsFormAction
     * @see processPaymentCountryRestrictionsFormAction
     * @see processPaymentCurrencyRestrictionsFormAction
     * @see processPaymentGroupRestrictionsFormAction
```
and deprecation error should probably be called in this method. (search for "@trigger_error" function in code will see examples with error_level E_USER_DEPRECATED

Reply:
I commited your text changes but I am not sure about trigger_error, this will still be used by PaymentPreferences if Feature flag for this is not enabled.